### ISSUE TITLE
[1.17.1] BlockEntity#onLoad is never called

### ISSUE THREAD

**Issue by malte0811** [MinecraftForge/MinecraftForge#7926]
**Minecraft Version:** 1.17.1

**Forge Version:** Tested on 37.0.2, confirmed in code in 6ddf56ef158af0282696f33d10ebed452a751375 (current latest 1.17 commit)

**Logs:** Not applicable

**Steps to Reproduce:**
 1. Implement `onLoad` in a block entity
 2. Observe that it is not called when the block entity is loaded

**Description of issue:**
The patch calling `BlockEntity#onLoad` seems to have been missed in the 1.17 update, there are no references to it in Forge besides the declaration in `IForgeBlockEntity`.

**Comment by XFactHD** [comment]
I have noticed this as well. After looking for a place to call `IForgeBlockEntity#onLoad` I came to the conclusion that this was probably not missed but rather no proper place was found yet due to the `CompleteableFuture` hell that is chunk loading.
I have found a spot that works for initial placement on client and server and is also called on the client after loading a chunk but finding a spot to call it server side on chunk load that isn't susceptible to deadlocks is tricky.
